packages/cli/src/nonInteractiveCli.ts

```diff
@@ -55,6 +55,30 @@ export async function runNonInteractive(
         ? new StreamJsonFormatter()
         : null;
 
+    const abortController = new AbortController();
+
+    // Track if we're in the process of aborting to handle multiple signals
+    let isAborting = false;
+
+    // Signal handler for graceful cancellation
+    const handleSignal = (signal: NodeJS.Signals) => {
+      if (isAborting) {
+        // Second signal: force exit immediately
+        process.stderr.write('\nForced exit\n');
+        process.exit(signal === 'SIGINT' ? 130 : 143);
+      }
+
+      isAborting = true;
+      process.stderr.write('\nCancelling...\n');
```

Comment:
I'd err on not showing this message or only showing this message if it has been 200ms and the terminal still hasn't cancelled.

Reply:
<img width="605" height="329" alt="Image" src="https://github.com/user-attachments/assets/d68b448d-fd94-46d6-aea1-ef7df3757e9f" />
output looks a bit verbose showing ^C, "Cancelling" and "Operation cancelled." in the output.